### PR TITLE
Don't notify that email addresses exist

### PR DESCRIPTION
**Pull request by andrebola**
issue #875

Note:
When a user changes the email for another which is already on the database we don't
inform the user that the email is already used

### CODE REVIEW

File: accounts/forms.py [hunk 1]
```diff
@@ -287,23 +268,6 @@ class EmailResetForm(forms.Form):
...
-    def clean_email(self):
-        email = self.cleaned_data["email"]
-        try:
-            get_user_by_email(email)
-            raise forms.ValidationError(_("A user using that email address already exists."))
-        except User.DoesNotExist:
-            pass
-        return email
-
-    def clean_password(self):
```
Comment (ffont) [inline accounts/forms.py]: Why is this clean function also removed?

File: accounts/views.py [hunk 2]
```diff
@@ -222,8 +222,14 @@ def resend_activation(request):
     if request.method == 'POST':
         form = ReactivationForm(request.POST)
         if form.is_valid():
-            user = form.cleaned_data['user']
-            send_activation(user)
+            username_or_email = form.cleaned_data['user']
```
Comment (ffont) [inline accounts/views.py]: So now activation works with usernames and emails right? This is a new feature not explained in the PR or issue description right?

File: accounts/views.py [hunk 3]
```diff
@@ -935,38 +946,45 @@ def old_user_link_redirect(request):
 @login_required
 def email_reset(request):
     if request.method == "POST":
-        form = EmailResetForm(request.POST, user=request.user)
```
Comment (ffont) [inline accounts/views.py]: This changes depend on whether `EmailResetForm` keeps the `clean_password` method (see my comment above).

File: accounts/forms.py (hunk 1 above)
Comment (andrebola) [reply accounts/forms.py]: I think this was removed by mistake, I will add it again

File: accounts/views.py (hunk 2 above)
Comment (andrebola) [reply accounts/views.py]: The clean_user method of the form used to check email and username, now this is made in the view.

**Comment by bdejong** [comment]
This is a relatively big change on user/email validation so I think this could really use a test so we know we're not breaking anything here.

**Comment by ffont** [comment]
I improved the tests and fixed a problem that was spotted after improving the tests. I'll merge the PR as now I'm quite positive it's ready to deploy.

**Pull request merged**